### DISCUSSION TITLE
Is Conditional Styling Possible?

### DISCUSSION THREAD

**Discussion by IIZard** [QuestPDF/QuestPDF#1065] [Q&A] [answered]
Hello,
I'm trying to add conditional styling to elements and it's getting quite tricky because of the requirement to use the fluent code structure.
Is there any way around this?

Example:-
I have a paragraph component, and a styling class which includes but is not limited to:
>An enum for HAllignment
>An enum for VAllignment
>Properties that describe various Sizes, Colours, Borders, Padding, etc..

How can I conditionally add these calls to an `Element` ?
Currently I get a `You should not assign multiple child elements to a single-child container.` exception when I try to do anything like this:

```
container.Element(e => {
    if(style.Size is not null){
        e.Width(style.Size.W);
        e.Height(style.Size.H);
    }
    .... etc
    return e;
});
```

How should I approach conditional styling in QuestPDF?

Any help would be appreciated. 
Thanks

**Comment by IIZard** [comment, marked as answer]
I've found a solution for this:-
I just write extension methods that return the element based on my class structure.
It looks similar to this:

```
container.Element(e => 
        e
        .HandleSize(style.Size)
        .HandleBackgroundColour(style.BackgrounColour)
        .... etc
    );

...

private static IContainer HandleSize(this IContainer container, Size? size){
    return size is null ? conatainer :
        container.Width(size.W).Height(size.H);
}

etc
```


Any better ways to do this or suggestions would still be appreciated.

**Reply by MarcinZiabek** [reply]
Your solution is correct. Depending on the context, you can also try:

```csharp
container
  .Element(c => style.BackgrounColour ? c.Background(style.BackgrounColour) : c)
  .MoreElements()
```

If you have any suggestions on how dynamic / custom / conditional styling could be handled by the Fluent API, please feel free to share 😄

**Comment by DawidRobertNowak** [comment]
Hello,

We are closing this thread due to inactivity. Thank you for taking the time to reach out and contribute to the project — your feedback helps us improve QuestPDF.

Should you encounter any further problems, please don't hesitate to open a new issue.

Thank you!
